playwright/src/main/java/com/microsoft/playwright/junit/impl/APIRequestContextExtension.java

```diff
@@ -0,0 +1,40 @@
+package com.microsoft.playwright.junit.impl;
+
+import com.microsoft.playwright.APIRequestContext;
+import com.microsoft.playwright.Playwright;
+import com.microsoft.playwright.junit.Options;
+import org.junit.jupiter.api.extension.*;
+
+import static com.microsoft.playwright.junit.impl.ExtensionUtils.isParameterSupported;
+
+public class APIRequestContextExtension implements ParameterResolver, AfterAllCallback {
+  private static final ThreadLocal<APIRequestContext> threadLocalAPIRequestContext = new ThreadLocal<>();
+
+  @Override
+  public void afterAll(ExtensionContext extensionContext) {
```

Comment:
This would mean that we use the same APIRequestContext across all tests in the file and we want to create a fresh one for each test (similar to BrowserContext). So we should either find a way to reset the context after all beforeAll hooks have finished and before the tests started or just not have APIRequestContext fixture in beforeAll hooks. Same applies to afterAll hooks. So basically beforeAll should have their own instance of APIRequestContext, each test should have its own and beforeAll hooks should have another instance.